src/commands/timelinecommands.cpp

```diff
@@ -109,18 +112,27 @@ void InsertCommand::redo()
             clip.set_in_and_out(info->frame_in, info->frame_out);
             bool lastClip = i == 0;
             m_model.insertClip(m_trackIndex, clip, m_position, m_rippleAllTracks, false, lastClip);
+            shift += clip.get_playtime();
         }
     } else {
         ProxyManager::generateIfNotExists(clip);
         m_model.insertClip(m_trackIndex, clip, m_position, m_rippleAllTracks, m_seek);
+        shift = clip.get_playtime();
     }
     m_undoHelper.recordAfterState();
+    if (m_rippleAllTracks && shift > 0) {
```

Comment:
Instead of using `m_rippleAllTracks` here, let's add a new member `m_rippleMarkers` and initialize it in the same way currently? Then, if decide make a distinct option for marker toggle, quite possibly in beta, it will be easier to change.